02_knight_and_princess/anton_potapov/graph_create.cpp

```diff
@@ -0,0 +1,115 @@
+#include <cassert>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "properties.hpp"
+
+using VertexId = int;
+using EdgeId = int;
+
+struct Vertex {
+  VertexId id;
+  Camp camp_info;
+  Vertex(int vertex_id, Camp c_info = Camp())
+      : id(vertex_id), camp_info(c_info) {}
+};
+
+struct Edge {
+  EdgeId id;
+  VertexId vertex1_id, vertex2_id;
+  Boss boss_info;
+  Edge(EdgeId e_id, VertexId ver1, VertexId ver2, Boss b_info = Boss())
```

Comment:
Я говорил фундаментальные типы, как `int`, `float`, и т.д. нужно передавать в функции по значению. Но так как мы объявили свой тип, который может быть изменен, и мы не уверены, `int` он или нет, то лучше передавать его как `const ref`
```suggestion
  Edge(const EdgeId& e_id, const VertexId& ver1, const VertexId& ver2, const Boss& b_info = Boss())
```